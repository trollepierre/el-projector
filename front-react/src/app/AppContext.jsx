import React, { useReducer, createContext, useContext } from 'react';
import PropTypes from 'prop-types';
import { reducer, initialState, actions } from './store';

const AppContext = createContext();

export const useAppContext = () => useContext(AppContext);

export const getProps = (state, dispatch) => ({
  ...state,
  setIsAuthenticated: isAuthenticated => actions.setIsAuthenticated(isAuthenticated)(dispatch),
  authenticate: password => actions.authenticate(password)(dispatch),
  loginSilently: () => actions.loginSilently()(dispatch),
});

export const AppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <AppContext.Provider value={{ ...getProps(state, dispatch) }}>
      {children}
    </AppContext.Provider>
  );
};

AppProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default AppContext;
