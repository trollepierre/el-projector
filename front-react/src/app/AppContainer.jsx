import React from 'react';
import { AppProvider, useAppContext } from './AppContext';

import LoginPage from '../pages/LoginPage';
import { Header } from '../bricks';
import ProjectPage from '../pages/ProjectPage/ProjectPage';

export const AppModule = () => {
  const { isAuthenticated } = useAppContext();

  if (!isAuthenticated) {
    return (<LoginPage/>);
  }

  return (
    <div className="App">
      <Header/>
      <ProjectPage/>
    </div>
  );
};

const AppContainer = () => (
  <AppProvider>
    <AppModule/>
  </AppProvider>
);

export default AppContainer;
