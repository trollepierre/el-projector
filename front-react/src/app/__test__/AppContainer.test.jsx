import React from 'react';
import { shallow } from 'enzyme';

import AppContainer, { AppModule } from '../AppContainer';
import * as AppContext from '../AppContext';

describe('AppContainer', () => {
  describe('AppModule', () => {
    it('should match snapshot when authenticated', () => {
      // Given
      jest
        .spyOn(AppContext, 'useAppContext')
        .mockImplementation(() => ({ isAuthenticated: true }));

      // When
      const wrapper = shallow(<AppModule/>);

      // Then
      expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot when not authenticated', () => {
      // Given
      jest
        .spyOn(AppContext, 'useAppContext')
        .mockImplementation(() => ({ isAuthenticated: false }));

      // When
      const wrapper = shallow(<AppModule/>);

      // Then
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('AppContainer', () => {
    it('should match snapshot', () => {
      // When
      const wrapper = shallow(<AppContainer/>);

      // Then
      expect(wrapper).toMatchSnapshot();
    });
  });
});
