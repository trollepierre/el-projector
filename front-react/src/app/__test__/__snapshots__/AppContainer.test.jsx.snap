// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer AppContainer should match snapshot 1`] = `
<AppProvider>
  <AppModule />
</AppProvider>
`;

exports[`AppContainer AppModule should match snapshot when authenticated 1`] = `
<div
  className="App"
>
  <Header />
  <ProjectPage />
</div>
`;

exports[`AppContainer AppModule should match snapshot when not authenticated 1`] = `<LoginPage />`;
